### ISSUE TITLE
Radeon VII PlaidML far slower than CPU Tensorflow

### ISSUE THREAD

**Issue by afogarty85** [plaidml/plaidml#1002]
I am looking for help to get the most out of my GPU for PlaidML ops. As of now, my CPU is killing the GPU in terms of performance, and the GPU is hardly being pushed when it is in use.

My environment is:
conda create -n my_ml python=3.7 jupyter numpy pandas matplotlib scikit-learn pip ipykernel tensorflow
pip install plaidml-keras plaidbench
plaidml-setup

With PlaidML, my keras version is: 2.2.4

With Tensorflow, my versions are:
tf: 2.1.0
keras: 2.2.4-tf

The results are consistent across batch sizes from 32 to 1024; CPU always win by far.

My code is:

import numpy as np
from sklearn.datasets import fetch_openml
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
import pandas as pd
import matplotlib.pyplot as plt
import time

import os
os.environ["KERAS_BACKEND"] = "plaidml.keras.backend"
import keras

# omit os/environ/import keras above when testing tensorflow cpu
#import tensorflow as tf
#from tensorflow import keras

tf.__version__
keras.__version__

X, y = fetch_openml(name='mnist_784', return_X_y = True, cache = False)
X = X / 255.0 
shuffle = np.random.permutation(np.arange(X.shape[0]))
X, y = X[shuffle], y[shuffle]
X = X.astype('float32')
y = y.astype('float32')

X_train_full, X_test, y_train_full, y_test = train_test_split(X, y, random_state = 42)
X_train, X_dev, y_train, y_dev = train_test_split(X_train_full, y_train_full, random_state = 42)

from functools import partial
RegularizedDense = partial(keras.layers.Dense,
                           activation = "elu",
                           kernel_initializer = "he_normal")

model = keras.models.Sequential([
    keras.layers.Dense(300, input_shape=[784]),
    keras.layers.Dropout(rate = 0.2),
    keras.layers.BatchNormalization(),
    RegularizedDense(300),
    keras.layers.Dropout(rate = 0.2),
    keras.layers.BatchNormalization(),
    RegularizedDense(100),
    keras.layers.Dropout(rate = 0.2),
    keras.layers.BatchNormalization(),
    RegularizedDense(10, activation = "softmax")
])

model.compile(loss = 'sparse_categorical_crossentropy', 
optimizer = keras.optimizers.Adam(lr = 5e-4, beta_1 = 0.9, beta_2 = 0.999), 
metrics = ['accuracy'])

start = time.time()
history = model.fit(X_train, y_train,
batch_size = 32,
epochs = 10, 
validation_data = (X_dev, y_dev)) 
end = time.time()
print(end - start)

**Comment by brianretford** [comment]
Generally speaking mnist is not a great target for GPU optimization, additionally pmlv0's performance on dense layers leaves something to be desired. The overall number of flops in mnist is very low and memory transfer times can easily dominate - see for instance https://github.com/tensorflow/tensorflow/issues/31654

**Comment by afogarty85** [comment]
Thank you for looking into the matter! Do you happen to have a stock problem that would stress the GPU hard that I might see if my installation is working as intended? I appreciate your time!

**Comment by brianretford** [comment]
Generally speaking plaidbench is a good way to test performance.
